tests/test_datasets.py

```diff
@@ -73,10 +73,21 @@ def test_make_classification_df():
         dates=(date(2014, 1, 1), date(2015, 1, 1)),
     )
 
+    X_df1, y_series1 = dask_ml.datasets.make_classification_df(
+        n_samples=100,
+        n_features=5,
+        random_state=123,
+        chunks=100,
+        dates=(date(2014, 1, 1), date(2015, 1, 1)),
+    )
+    check_randomness = np.unique((X_df["date"] == X_df1["date"]).compute())
```

Comment:
Yes, the code I've written checks for repeatability, on account of the seed. Since the the numpy's `randint` function is a deterministic pseudo random number generator, we can be sure that it will produce a random set of numbers. I've ensured in the code that the function `random_date` is called multiple times, each with a different seed. Hence I feel the line `np.unique(X["date"]).size >= threshold` would be redundant. Open to any thoughts you might have here @stsievert

Reply:
The same `random_seed=123` is passed to both calls to `make_classification_df`; I would expect every value in `X_df["date"]` and `X_df1["date"]` to be the same.

I think `X["date"].nunique() >= threshold` would be a lot simpler.

Reply:
Okay, that sounds good. Now I've to figure out, what would be a good threshold. Will `n_samples/2` be good?

Reply:
Oh yeah, `threshold=n_samples/2` is more than good. I think `threshold=2` would suffice; that'd make sure #845 is resolved.